src/Model/FileSupport/BuyRequest/CustomizableOptionDataProvider.php

```diff
@@ -0,0 +1,198 @@
+            if ($optionType == self::OPTION_TYPE) {
+                if ($fileData !== false) {
+                    $this->createFileAndFolder($option->getUid(), $fileData['raw'], $fileData['title']);
+                    unset($fileData['raw']);
+                    $customizableOptionsData[$optionId][] = $fileData;
+            // -- End of Support --
+                } else {
+                    $customizableOptionsData[$optionId][] = $option->getValue();
+                }
+            }
+        }
+
+        $output = ['options' => $this->flattenOptionValues($customizableOptionsData)];
+        return $output;
+    }
+
+    private function getFileData($uid, $optionData) {
+        try {
+            $data = json_decode($optionData, true);
+            if (!is_array($data)) {
+                return false;
+            }
+
+            $filename = $data['file_name'];
+            $filedata = $data['file_data'];
+
+            $insidePath = $uid . '/_/' . $filename;
+            if (!$filename || !$filedata) {
+                return false;
+            }
+
+            return [
+                'type' => 'application/octet-stream',
+                'title' => $filename,
+                'quote_path' => self::QUOTE_MEDIA_PATH . $insidePath,
+                'order_path' => self::ORDER_MEDIA_PATH . $insidePath,
+                'fullpath' => $this->mediaPath . self::QUOTE_MEDIA_PATH . $insidePath,
+                'secret_key' => $filename,
+                'raw' => $filedata
+            ];
+        } catch(\Exception $e) {
+            return false;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param $quoteId
+     * @param $value
+     * @param $filename
+     */
+    public function createFileAndFolder($uid, $value, $filename)
+    {
+        $directory = sprintf(
+            '%s%s%s/_',
+            $this->mediaPath,
+            self::QUOTE_MEDIA_PATH,
+            $uid
+        );
+
+        if (!is_dir($directory)) {
+            mkdir($directory, 0777, true);
+        }
+
+        file_put_contents(
+            sprintf('%s/%s', $directory, $filename),
+            base64_decode(substr($value, strpos($value, ',') + 1 ))
+        );
+    }
+
+    /**
+     * Flatten option values for non-multiselect customizable options
+     *
+     * @param array $customizableOptionsData
+     * @return array
+     */
+    private function flattenOptionValues(array $customizableOptionsData): array
+    {
+        foreach ($customizableOptionsData as $optionId => $optionValue) {
+            if (count($optionValue) === 1) {
+                $customizableOptionsData[$optionId] = $optionValue[0];
+            }
+        }
+
+        return $customizableOptionsData;
+    }
+
+    /**
+     * Checks whether this provider is applicable for the current option
+     *
+     * @param array $optionData
+     * @return bool
+     */
+    private function isProviderApplicable(array $optionData): bool
```

Comment:
Is this necessary to be private ?

If no then please make everything protected over private until it is absolutely necessary. We are extensibility-first organisation, and private stuff breaks the concept.